src/test/signupValidationTest.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable import/no-extraneous-dependencies */
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import index from '../index';
+
+chai.use(chaiHttp);
+chai.should();
+
+describe('SignupValidationErrors', () => {
+  describe('POST/', () => {
+    it('bad input because of bad email', (done) => {
+      const data = {
+        username: 'kagabo',
+        password: 'Fofo1',
+        email: 'fofo@gmail.com'
+      };
+      chai.request(index)
+        .post('/api/users')
+        .send(data)
+        .end((req, res) => {
+          res.should.have.status(400);
+          res.body.err[0].should.be.eql('password is required and should look like: Example1@');
+          res.should.be.json;
+          done();
+        });
+    });
+
+    it('bad password on signout', (done) => {
+      const data = {
+        username: 'kagabo',
+        password: 'Fofo1@',
+        email: 'fofogmail.com'
+      };
+      chai.request(index)
+        .post('/api/users')
+        .send(data)
+        .end((req, res) => {
+          res.should.have.status(400);
+          res.body.err[0].should.be.eql('Email is required and should look like: example@example.com!');
+          res.should.be.json;
+          done();
+        });
+    });
+
+    it('bad email and password on signup', (done) => {
+      const data = {
+        username: 'kagabo',
+        password: 'Fofo1',
+        email: 'fofogmail.com'
+      };
+      chai.request(index)
+        .post('/api/users')
+        .send(data)
+        .end((req, res) => {
+          res.should.have.status(400);
+          res.body.err[0].should.be.eql('Email is required and should look like: example@example.com!');
+          res.body.err[1].should.be.eql('password is required and should look like: Example1@');
+          res.should.be.json;
+          done();
+        });
+    });
+  });
+});
```

Comment:
I think those tests for signup validations should in the user.js file with other tests related to the user.

Reply:
that true.